src/compiler/binder.ts

```diff
@@ -410,13 +410,17 @@ namespace ts {
          * @param includes - The SymbolFlags that node has in addition to its declaration type (eg: export, ambient, etc.)
          * @param excludes - The flags which node cannot be declared alongside in a symbol table. Used to report forbidden declarations.
          */
-        function declareSymbol(symbolTable: SymbolTable, parent: Symbol | undefined, node: Declaration, includes: SymbolFlags, excludes: SymbolFlags, isReplaceableByMethod?: boolean): Symbol {
-            Debug.assert(!hasDynamicName(node));
+        function declareSymbol(symbolTable: SymbolTable, parent: Symbol | undefined, node: Declaration, includes: SymbolFlags, excludes: SymbolFlags, isReplaceableByMethod?: boolean, isComputedName?: boolean): Symbol {
+            Debug.assert(isComputedName || !hasDynamicName(node));
 
             const isDefaultExport = hasSyntacticModifier(node, ModifierFlags.Default) || isExportSpecifier(node) && node.name.escapedText === "default";
 
             // The exported symbol for an export default function/class node is always named "default"
-            const name = isDefaultExport && parent ? InternalSymbolName.Default : getDeclarationName(node);
+            const name = isComputedName

```

Comment:
would it be correct to do this for any node like so? 
```suggestion
            const name = hasDynamicName(node)
```

The assert at the top is suggestive.

Reply:
I chose not to remove the debug since `declaredMethod` is called on multiple places unrelated to this bug. I'm not sure what other side effects can I cause by removing it.

If you feel strongly on this one I don't mind changing it tho.

Reply:
I'm pretty sure declareSymbol isn't currently called for any dynamic name, and with this change will now be called for some dynamic names. I think it's better to support all dynamic names the same as non-dynamic ones, in case future uses of dynamic names need to declare symbols.